pkl-tools/pkl-tools.gradle.kts

```diff
@@ -65,6 +65,58 @@ val javadocJar by
 
 tasks.shadowJar { archiveBaseName.set("pkl-tools-all") }
 
+private fun Exec.configureTestStartFatJar(launcher: Provider<JavaLauncher>) {
+  dependsOn(tasks.shadowJar)
+  group = "verification"
+
+  // placeholder output to satisfy up-to-date check
+  val outputFile = layout.buildDirectory.file("testStartFatJar/${name}.txt")
+  inputs.files(tasks.shadowJar)
+  executable = launcher.get().executablePath.asFile.absolutePath
+
+  argumentProviders.add(
+    CommandLineArgumentProvider {
+      buildList {
+        add("-cp")
+        add(tasks.shadowJar.get().outputs.files.singleFile.absolutePath)
+        add("org.pkl.cli.Main")
+        add("eval")
+        add("-x")
+        add("1 + 1")
+        add("pkl:base")
+      }
+    }
+  )
+
+  doFirst { outputFile.get().asFile.delete() }
```

Comment:
Why do you do this? 🤔 If the input changes the task run. But the output can be stable no? 🤔

Reply:
Yeah, you're right; no need to for this line.